src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanFL401ValidationService.java

```diff
@@ -0,0 +1,229 @@
+                    errorItems,
+                    String.format(
+                            "Section %s - Applicant must have a relationship with the "
+                                    + "respondent",
+                            sectionName),
+                    Status.ERRORS);
+        }
+    }
+
+    private int applicantRespondentRelationshipCounter(
+            Map<String, String> ocrApplicantRespondentFieldsMap) {
+        int relationshipCounter = 0;
+
+        if (null != ocrApplicantRespondentFieldsMap && !ocrApplicantRespondentFieldsMap.isEmpty()) {
+            for (Map.Entry<String, String> applicantRespondentRelationshipField :
+                    ocrApplicantRespondentFieldsMap.entrySet()) {
+                if (isValidRelationshipApplicantRespondent(
+                        ocrApplicantRespondentFieldsMap,
+                        applicantRespondentRelationshipField.getKey(),
+                        YES)) {
+                    relationshipCounter++;
+                }
+            }
+        }
+
+        return relationshipCounter;
+    }
+
+    private boolean isValidRelationshipApplicantRespondent(
+            Map<String, String> ocrApplicantRespondentFieldsMap,
+            String fieldName,
+            String fieldValidationVal) {
+
+        return ocrApplicantRespondentFieldsMap.containsKey(fieldName)
+                && ocrApplicantRespondentFieldsMap
+                        .get(fieldName)
+                        .equalsIgnoreCase(fieldValidationVal);
+    }
+
+    private void setErrorWarningMsg(
+            BulkScanValidationResponse bulkScanValidationResponse,
+            List<String> items,
+            String fieldMsg,
+            Status status) {
+
+        if (status.equals(Status.ERRORS)) {
+            items.add(fieldMsg);
+            bulkScanValidationResponse.addErrors(items);
+            bulkScanValidationResponse.setStatus(status);
+        } else if (status.equals(Status.WARNINGS)) {
+            items.add(fieldMsg);
+            bulkScanValidationResponse.addWarning(items);
+            bulkScanValidationResponse.setStatus(status);
+        }
+
+        bulkScanValidationResponse.changeStatus();
+    }
+
+    private Map<String, String> getApplicantRespondentRelationshipFieldsMap(
+            Map<String, String> ocrDataFields, String relationshipRegexField) {
+
+        TreeMap<String, String> relationshipMap = new TreeMap<>();
+
+        if (null != ocrDataFields && !ocrDataFields.isEmpty()) {
+            for (Map.Entry<String, String> applRespRelationFieldMap : ocrDataFields.entrySet()) {
+                if (applRespRelationFieldMap.getKey().matches(relationshipRegexField)) {
+                    relationshipMap.put(
+                            applRespRelationFieldMap.getKey(), applRespRelationFieldMap.getValue());
+                }
+            }
+        }
+
+        return relationshipMap;
+    }
+
+    private List<String> validateInputDate(
```

Comment:
it's duplicate method. either make this public and removed from FL401 service class or delete this one and make FL401 Service method public.

Reply:
Not duplicate as having different signatures. Method renamed.